hearth/lib/hearth/http/request.rb

```diff
@@ -7,77 +7,75 @@ module Hearth
   module HTTP
     # Represents an HTTP request.
     # @api private
-    class Request
+    class Request < Hearth::Request
+      # @param [URI, String] uri The URI. Also sets :destination.
       # @param [String] http_method
-      # @param [String] url
-      # @param [Headers] headers
-      # @param [IO] body
-      def initialize(http_method: nil, url: nil, headers: Headers.new,
-                     body: StringIO.new)
+      # @param [Fields] fields
+      # @param (see Hearth::Request#initialize)
+      def initialize(uri:, http_method: nil, fields: Fields.new, **kwargs)
+        kwargs[:destination] = URI(uri)
```

Comment:
This is a little weird with inheritance.  You cannot create an `HTTPRequest` with destination as a keyword, but uri is required.  And the type of destination changes depending on whether this is a parent or child class (and String and URI are not interchangable - so this means you can't use an HTTPRequest in place of a Request ie Liskov Substitution Principle) Feels like we're trying to have it both ways and I think its making it a bit clunkier.  

I'd lean towards either:
1.  Eliminate `uri` as an httpRequest concept and just stick with the SRA's `destination`.
2.  the SRA lists Request's `destination` as a URI - so do that rather than a String.  That simplifies this constructor a bit and then we can optionally take uri OR destination here.
3. just use uri for both cases (since again the SRA lists the type of destination in Request as URI).